lib/Statocles/App/Perldoc.pm

```diff
@@ -212,12 +214,38 @@ sub pages {
                 crumbtrail => \@crumbtrail,
             },
         );
-
+        # unable to highlight source as source.html.ep uses <%== %> to escape html.
+        # $self->_highlight_page( $pages[-1], 'pre' );
     }
 
     return @pages;
 }
 
+sub _highlight_page {
+  my ( $self, $page, $sel ) = @_;
+  # highlight only if site-wide highlighting is available
+  return unless my $hl = $page->site->plugins->{highlight};
+  # only highlight the page content
+  my $dom = Mojo::DOM->new($page->content);
```

Comment:
This was my first approach, but the `render` method used in the tests did not return the edits.

Reply:
Ah, no. `render` is used to build the Mojo::DOM object, and then that object is used the rest of the time. So, it'd be better to test that the DOM has the changes you expect.

Reply:
I'm not sure I fully understand the distinction you are making here. I was referring to [Statocles/Test.pm#L206-L227](https://github.com/preaction/Statocles/blob/master/lib/Statocles/Test.pm#L206-L227).

Reply:
Right, that's a bug: It's doesn't accurately reflect the documentation and intention of the Mojo::DOM object being the canonical representation of the page. That's totally my fault, so I am currently fixing that method you linked to and fixing the bugs that will inevitably result (and finally perhaps fix the other test failure that has been plaguing me recently...)

Reply:
Okay. That's changed in 14ed8ddb127b8bb2ad64fa768ec82e75e4e22fdc so you should be able to use the `dom` attribute as I had intended. Sorry about the rigamarole.

Reply:
No worries on the misunderstandings, these things happen at distance. I now see the `My::Test::_Extras` has a `test_pages`, just like, but not like `Statocles::Test::test_pages`.

`Statocles::Test::test_pages is deprecated and will be removed in v1.000. See Statocles::Help::Upgrading`, but I cannot see a reference to the deprecated method in the help. Can you advise which to use? I think [it](https://github.com/preaction/Statocles/blob/master/t/lib/My/Test.pm#L158) needs fixing too.

Reply:
Well, that would explain why it was so easy to fix: I fixed the code that isn't being used...

Reply:
Alright, so that's fixed, and I really ought to start planning out the v1 changes I want to make so I can start pushing that and actually removing deprecated stuff...

Reply:
Awesome, working as expected now in b03502b.